src/PhpSpreadsheet/Calculation/Financial/Coupons.php

```diff
@@ -261,6 +261,7 @@ public static function COUPNCD(
             self::validateCouponPeriod($settlement, $maturity);
             $frequency = FinancialValidations::validateFrequency($frequency);
             $basis = FinancialValidations::validateBasis($basis);
```

Comment:
Would
```php
$basis = (bool) FinancialValidations::validateBasis($basis);
```
be better here? No method call overhead to cast it to a boolean

Reply:
This was a response to cases where Scrutinizer was correct (in a nagging it-really-doesn't matter way). It says $basis is unused, so the assignment is dead code. Casting won't help. And, if you remove the statement altogether, it will complain that the parameter $basis is unused. We are in a bind; Excel accepts the parameter, so we have to accept it whether or not our code needs it. I tried annotating the parameter, but one of our scanners (can't remember if it was phpcs or php-cs-fixer) objected. This seemed like a cheap way to keep everyone happy. I intend to revisit all of these eventually. For now, I'm just looking for a way to eliminate problems like this so that it is easier for me to run through the Scrutinizer issues to find complaints which are legitimate (there are some of those) without having to continually wade through things I've already decided are harmless.